middlewares/prometheus_test.go

```diff
@@ -44,4 +46,24 @@ func TestPrometheus(t *testing.T) {
 	if !strings.Contains(body, latencyName) {
 		t.Errorf("body does not contain request duration entry '%s'", reqsName)
 	}
+
+	// Register the same metrics again
+	metricsMiddlewareBackend = NewMetricsWrapper(NewPrometheus("test", &types.Prometheus{}))
+	n = negroni.New()
+	n.Use(metricsMiddlewareBackend)
+	n.UseHandler(r)
+
+	recorder = httptest.NewRecorder()
+	n.ServeHTTP(recorder, req2)
+
+	body = recorder.Body.String()
+	re := regexp.MustCompile(`(?m)traefik_requests_total.*?(\d)$`)
+	if re.FindStringSubmatch(body)[1] != "2" {
+		t.Error("body does not contain correct value for total requests")
+	}
+
+	metrics, _ := prometheus.DefaultGatherer.Gather()
+	if metrics[len(metrics)-1].Metric[0].GetCounter().GetValue() != 3 {
+		t.Error("gathered metrics does not contain correct value for total requests")
+	}
```

Comment:
Can we extend the test to also cover the Histogram tracking the request duration? I think we can use the [Histogram `SampleCount`](https://github.com/prometheus/client_model/blob/6f3806018612930941127f2a7c6c453ba2c527d2/go/metrics.pb.go#L203) to assert on.